server/src/main/java/org/elasticsearch/search/sort/SortValue.java

```diff
@@ -299,10 +299,10 @@ private BytesSortValue(StreamInput in) throws IOException {
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            if (out.getVersion().before(Version.V_7_11_0)) {
+            if (out.getTransportVersion().before(TransportVersion.V_7_11_0)) {
                 throw new IllegalArgumentException(
-                    "versions of Elasticsearch before 7.11.0 can't handle non-numeric sort values and attempted to send to ["
-                        + out.getVersion()
+                    "data versions before [7110099] can't handle non-numeric sort values, attempted to send to ["
```

Comment:
perhaps `transport` instead of `data` ?

Reply:
I'm wondering if 'transport' is a bit of an internal term, users who might see this exception will understand 'data' more. But we need to go over exception messages anyway...

Reply:
I think we should leave the "transport" terminology. Data does not have a concrete meaning (and could very well be confused with index data, a loose term that is also used). I don't think these messages are possible in real user scenarios. These normally appear in transport code to guard against usage of a new feature while in a mixed cluster. IMO this is an unlikely scenario because it means a user has changed their client to begin using a new feature while in the middle of a deployment. I don't know of any developers who would do such a thing, it doesn't make sense. So I don't think we should be worried about these messages' wording.

Reply:
Updated to transport